src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+
+import { TelemetryService } from 'services';
+import { DeviceIcon } from './deviceIcon';
+import { RulesGrid, rulesColumnDefs } from 'components/pages/rules/rulesGrid';
+import { translateColumnDefs } from 'utilities';
+import {
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  FlyoutSection,
+  SectionHeader,
+  SectionDesc,
+  BasicGrid as Grid,
+  BasicRow as Row,
+  BasicHeaderCell as HeaderCell,
+  BasicCell as Cell
+} from 'components/shared';
+
+import './deviceDetails.css';
+
+export class DeviceDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { alarms: undefined, isAlarmsPending: false, alarmsError: undefined };
+    this.columnDefs = [
+      rulesColumnDefs.ruleName,
+      rulesColumnDefs.severity,
+      rulesColumnDefs.explore
+    ];
+  }
+
+  componentDidMount() {
+    if (!this.props.rulesLastUpdated) this.props.fetchRules();
+    this.fetchAlarms((this.props.device || {}).id)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.fetchAlarms((nextProps.device || {}).id)
+  }
+
+  applyRuleNames = (alarms, rules) =>
+    alarms.map(alarm => ({
+      ...alarm,
+      name: (rules[alarm.ruleId] || {}).name
+    }));
+
+  fetchAlarms = (deviceId) => {
+    this.setState({ isAlarmsPending: true });
+    TelemetryService.getAlarms({
+      limit: 5,
+      order: "desc",
+      devices: deviceId
+    })
+      .subscribe(
```

Comment:
On thing to note is that whenever a subscribe occurs, it returns a subscription object. You need to save that object and call `.unsubscribe` on it in `componentWillUnmount()`. If a user closes the flyout before the call completes and the subscription isn't unsubscribed then the onNext or onError function will be called. Those will try to call `setState` on a component that doesn't exist anymore, causing an error.